Unix/create_user.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/sh
+#
+# Script to create user account and password for use by OMI
+# This file should be executed via ".", as it sets environment variables
+
+username=$OMI_USER
+
+if [ "x$username" = "x" ]; then
+    echo "Creating new user..."
+    username=omi_$USER
+fi
+
+egrep "^$username" /etc/passwd >/dev/null
+if [ $? -eq 0 ]; then
+    echo "User $username exists!"
+else
+    useradd=`which useradd`
+    if [ "x$useradd" = "x" ]; then
+        echo "Failed to find 'useradd'"
+        exit 1
+    fi
+
+    sudo $useradd -m $username
+    [ $? -eq 0 ] && echo "User $username has been added to system!" || echo "Failed to add a user!"
+fi
+
+userpasswd=`cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 28 | head -n 1`
+if [ "x$userpasswd" = "x" ]; then
+    echo "Failed to generate password"
+    exit 1
+fi
+
+echo $username:$userpasswd | sudo chpasswd
+[ $? -eq 0 ] && echo "User password has been changed!" || echo "Failed to change password!"
```

Comment:
If failures occur, you just continue in-line. Is that really what you want to do here? Shouldn't we abort if, for example, the account could not be created?